synapse/common.py

```diff
@@ -1238,6 +1238,30 @@ async def wait_for(fut, timeout):
     async with _timeout(timeout):
         return await fut
 
+async def waitretn(futu, timeout):
```

Comment:
From a code tidiness perspective, we'll need to move this **after** the asyncio vendored code for implementing `wait_for`.

Reply:
I'm not sure I understand, the function `waitretn` is after the custom implementation of `wait_for`. Did you want me to add a comment that is depends on it so we can review on the py3.12+ upgrade?